src/wayland-egldisplay.c

```diff
@@ -670,48 +763,74 @@ EGLBoolean wlEglTerminateHook(EGLDisplay dpy)
     return res;
 }
 
-static void getServerProtocolsInfo(struct wl_display *nativeDpy,
+static bool getServerProtocolsInfo(struct wl_display *nativeDpy,
                                    WlServerProtocols *protocols)
 {
     struct wl_display     *wrapper      = NULL;
     struct wl_registry    *wlRegistry   = NULL;
     struct wl_event_queue *queue        = wl_display_create_queue(nativeDpy);
     int                    ret          = 0;
+    bool                   result       = false;
     const struct wl_registry_listener registryListener = {
         registry_handle_global_check_protocols,
         registry_handle_global_remove
     };
 
     if (queue == NULL) {
-        return;
+        goto done;
     }
 
     wrapper = wl_proxy_create_wrapper(nativeDpy);
+    if (wrapper == NULL) {
+        goto done;
+    }
     wl_proxy_set_queue((struct wl_proxy *)wrapper, queue);
 
     /* Listen to wl_registry events and make a roundtrip in order to find the
      * wl_eglstream_display global object.
      */
     wlRegistry = wl_display_get_registry(wrapper);
-    wl_proxy_wrapper_destroy(wrapper); /* Done with wrapper */
+    if (wlRegistry == NULL) {
+        goto done;
+    }
     ret = wl_registry_add_listener(wlRegistry,
                                    &registryListener,
                                    protocols);
     if (ret == 0) {
         wl_display_roundtrip_queue(nativeDpy, queue);
-        if (protocols->hasDrm) {
+        if (protocols->wlDmaBuf) {
+            struct zwp_linux_dmabuf_feedback_v1 *default_feedback
+                    = zwp_linux_dmabuf_v1_get_default_feedback(protocols->wlDmaBuf);
+            if (default_feedback) {
+                zwp_linux_dmabuf_feedback_v1_add_listener(default_feedback, &dmabuf_feedback_check_listener, protocols);
+                wl_display_roundtrip_queue(nativeDpy, queue);
+                zwp_linux_dmabuf_feedback_v1_destroy(default_feedback);
+            }
+        } else if (protocols->wlDrm) {
+            wl_drm_add_listener(protocols->wlDrm, &drmListener, protocols);
             wl_display_roundtrip_queue(nativeDpy, queue);
-            /* destroy our wl_drm object */
-            wl_drm_destroy(protocols->wldrm);
+        }
+        result = protocols->drm_name != NULL;
```

Comment:
I see some discussion of this, but is this really fatal? I thought the streams path worked fine without this.

Reply:
I think it kinda doesn't matter since `checkNvidiaDrmDevice` will return false if `protocols->drm_name` is NULL, meaning we will fail `eglGetPlatformDisplay` anyway unless `__NV_PRIME_RENDER_OFFLOAD` is set. But PRIME render-offload definitely doesn't doesn't work with the EGLStreams path.

Reply:
Hrm. Did we regress that as part of the not-waking-up-GPUs-for-every-eglGetDisplay() change? It seems like if so, we sort of broke the eglstreams path. IIRC the wl_drm usage was added for the dma-buf path, so I doubt the eglstreams path was intended to require it.

Reply:
The EGLStreams path only works if the compositor calls `eglBindWaylandDisplayWL`, since that's where we bind the `wl_eglstream` interface. But in that case we will also bind our own implementation of `wl_drm`. So we shouldn't really need to worry about the EGLStreams path working without `wl_drm`, it will always be available.

To your question, though, no I don't think the change you mentioned regressed anything. All it did was move the failure point a little bit earlier. But even before that change, if we couldn't determine what device the compositor was using we would  fail (unless `__NV_PRIME_RENDER_OFFLOAD` is set).

Reply:
OK. perhaps I'm misremembering.

Reply:
I mean, there *was* a time when we would just choose a device completely arbitrarily, with no concern for which device the compositor was using. But we needed to change that because it was problematic on PRIME systems. See https://github.com/NVIDIA/egl-wayland/commit/d4937adc5cd04ac7df98fc5616e40319fb52fdee